src/signer.rs

```diff
@@ -1,50 +1,127 @@
 use core::marker::PhantomData;
 
-use hashbrown::HashMap;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{pallet_prelude::RuntimeDebug, traits::Get, BoundedBTreeMap};
+use frame_system::Config as SysConfig;
 use move_core_types::account_address::AccountAddress;
+use scale_info::TypeInfo;
 
-use crate::{balance::BalanceOf, Config, Error, Pallet};
+use crate::{
+    balance::{BalanceAdapter, BalanceOf},
+    Config, Error, Pallet,
+};
+
+/// Generic definition of the hash of an extrinsic call.
```

Comment:
Maybe mention that this is literally the hash of the TransactionTx for the execute extrinsic since `the hash of an extrinsic call` doesn't sound very specific

Reply:
Done, please check again :blush: